test/mpi/Makefile.mtest

```diff
@@ -20,7 +20,7 @@ LDADD = $(top_builddir)/util/mtest.$(OBJEXT) $(top_builddir)/util/mtest_common.$
 
 # Add libdtpools support
 AM_CPPFLAGS += -I$(top_srcdir)/dtpools/include
-LDADD += $(top_builddir)/dtpools/src/libdtpools.la
+LDADD += $(top_builddir)/dtpools/src/libdtpools.la @threadlib@
```

Comment:
We have this already in `Makefile_threads.mtest`, so I'm a little confused about adding it here. Do all tests depend on Argobots in this scenario? I.e. not just the tests that create threads?

Reply:
Good question. I'll investigate more.

Reply:
Yes. All tests depend on argonbots when the configure options is passed in, due to these lines in `mtest.c`: https://github.com/pmodels/mpich/blob/cb944baeb07061759f7c22d85704cc9c673056f9/test/mpi/util/mtest.c#L72